hummingbird/ml/operator_converters/linear_classifier.py

```diff
@@ -0,0 +1,98 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+import torch
+import numpy as np
+from onnxconverter_common.registration import register_converter
+
+"""
+Converters for scikit-learn linear models: LinearRegression, LogisticRegression, LinearSVC, SGDClassifier, LogisticRegressionCV
+"""
+
+
+class SklearnLinearModel(torch.nn.Module):
+    def __init__(self, coefficients, intercepts, classes, multi_class, device, is_linear_regression=False):
+        super(SklearnLinearModel, self).__init__()
+        self.coefficients = torch.nn.Parameter(torch.from_numpy(coefficients), requires_grad=False)
+        self.intercepts = torch.nn.Parameter(torch.from_numpy(intercepts), requires_grad=False)
+        self.classes = torch.nn.Parameter(torch.IntTensor(classes), requires_grad=False)
+        self.multi_class = multi_class
+        self.regression = is_linear_regression
+
+        self.perform_class_select = False
+        if min(classes) != 0 or max(classes) != len(classes) - 1:
+            self.perform_class_select = True
+
+        self.binary_classification = False
+        if len(classes) == 2:
+            self.binary_classification = True
+
+    def forward(self, x):
+        if self.multi_class == "multinomial":
+            output = torch.softmax(torch.addmm(self.intercepts, x, self.coefficients), dim=1)
+        elif self.regression:
+            output = torch.addmm(self.intercepts, x, self.coefficients)
+            if not self.binary_classification:
+                return output
+        else:
+            output = torch.sigmoid(torch.addmm(self.intercepts, x, self.coefficients))
+            if not self.binary_classification:
+                output /= torch.sum(output, dim=1, keepdim=True)
+
+        if self.binary_classification:
+            output = torch.cat([1 - output, output], dim=1)
+
+        if self.perform_class_select:
+            return torch.index_select(self.classes, 0, torch.argmax(output, dim=1)), output
+        else:
+            return torch.argmax(output, dim=1), output
+
+
+def convert_sklearn_linear_model(operator, device, extra_config):
```

Comment:
should we have a different method for linear regression? It looks that the code for linear regression is quite different than the other operators. In this way we can avoid to have to understand which branch will be taken and just explicitly spell out the conversion.

Reply:
please take another look? ok now?